zellij-server/src/screen.rs

```diff
@@ -2815,6 +2823,36 @@ pub(crate) fn screen_thread_main(
                 }
                 screen.report_tab_state()?;
             },
+            ScreenInstruction::RequestPluginPermissions(plugin_id, permissions) => {
+                let permissions = match GrantedPermission::from_default() {
+                    Ok(granted_permission) => {
+                        if let Some(p) = granted_permission.get(&permissions.name) {
+                            // TODO: How to deal cached data?
+                            permissions
+                        } else {
+                            permissions
+                        }
+                    },
+                    Err(_) => permissions,
+                };
```

Comment:
@imsnif 

Now, we don't save permission data in memory. Permission data is fetched from the cached file on every request.

But, There is still one dilemma here. If the cache data is fetched and applied, there is no way for the user to see the Request UI again. The only way is clearing the cache data.

Otherwise, if the cache data is not imported and used, the cache is meaningless.

Is there any good solution?

Reply:
Hmm... I see what you mean. Good catch.

Do you think we can:
1. If the user granted permissions, we remember it until they delete the cache explicitly
2. If the user did not grant permissions, we ask them again (if the plugin requests permissions) once per session

What do you think?

Reply:
Actually, it would be even better (if it's not too much work), to also do:

3. If the user presses `Ctrl c`, it is "no and never ask me again"

(I can add the UI for this)

But really, only if it's not a lot more work for you!

Reply:
Oh Thank you! it makes sense.

> If the user presses Ctrl c, it is "no and never ask me again"

Yes, this way would be better. But, for now, I will implement it based on 1 and 2. I think 3 would be better to improve later.